crates/prune/src/pruner.rs

```diff
@@ -185,7 +192,7 @@ impl<DB: Database> Pruner<DB> {
                     .set(to_block as f64);
 
                 done = done && output.done;
-                delete_limit = delete_limit.saturating_sub(output.pruned);
+                deleted += 1;
```

Comment:
I think this is incorrect: `deleted` means here the number of pruned segments, but we use it as a number of deleted entries in the `limit.at_limit`.

I think we should rather adjust the `PruneLimit` deletion limit budget when we finish pruning the segment.

Reply:
on main, `delete_limit` only exists within scope of this function. I think the change that needs to happen is rename the `entries_limit` to `items_limit` to encompass both segments and entries. the 2 layers basically, where entries are nested in segments.

Reply:
now I see what you mean. my bad, I was really hungry.